scenario/utils/relayPolygonMessage.ts

```diff
@@ -1,53 +1,134 @@
 import { impersonateAddress } from '../../plugins/scenario/utils';
 import { setNextBaseFeeToZero, setNextBlockTimestamp } from './hreUtils';
-import { Event } from 'ethers';
+import { Contract, Event, ethers } from 'ethers';
 
-export default async function relayMumbaiMessage(
+type BridgeERC20Data = {
+  syncData: string;
+  user: string;
+  rootToken: string;
+  amount: bigint;
+};
+
+function tryDecodeStateSyncedData(stateSyncedData: any): BridgeERC20Data | undefined {
+  try {
+    const { syncData } = ethers.utils.defaultAbiCoder.decode(
+      ['bytes32', 'bytes syncData'],
+      stateSyncedData
+    );
+    const { user, rootToken, depositData } = ethers.utils.defaultAbiCoder.decode(
+      ['address user', 'address rootToken', 'bytes depositData'],
+      syncData
+    );
+    const { amount } = ethers.utils.defaultAbiCoder.decode(['uint256 amount'], depositData);
+    return {
+      syncData,
+      user,
+      rootToken,
+      amount
+    };
+  } catch (e) {
+    return undefined;
+  }
+}
+
+export default async function relayPolygonMessage(
   governanceDeploymentManager: DeploymentManager,
   bridgeDeploymentManager: DeploymentManager,
+  startingBlockNumber: number
 ) {
-  const MUMBAI_RECEIVER_ADDRESSS = '0x0000000000000000000000000000000000001001';
-  const EVENT_LISTENER_TIMEOUT = 60000;
+  const POLYGON_RECEIVER_ADDRESSS = '0x0000000000000000000000000000000000001001';
+  const CHILD_CHAIN_MANAGER_PROXY_ADDRESS =
+    bridgeDeploymentManager.network === 'polygon'
+      ? '0xA6FA4fB5f76172d178d61B04b0ecd319C5d1C0aa'
+      : '0xb5505a6d998549090530911180f38aC5130101c6';
 
   const stateSender = await governanceDeploymentManager.getContractOrThrow('stateSender');
   const bridgeReceiver = await bridgeDeploymentManager.getContractOrThrow('bridgeReceiver');
   const fxChild = await bridgeDeploymentManager.getContractOrThrow('fxChild');
+  const childChainManager = new Contract(
+    CHILD_CHAIN_MANAGER_PROXY_ADDRESS,
+    [
+      'function rootToChildToken(address rootToken) public view returns (address)',
+      'function onStateReceive(uint256, bytes calldata data) external',
+      'function DEPOSIT() public view returns (bytes32)'
+    ],
+    bridgeDeploymentManager.hre.ethers.provider
+  );
 
-  // listen on events on the fxRoot contract
-  const stateSyncedListenerPromise = new Promise((resolve, reject) => {
-    const filter = stateSender.filters.StateSynced();
-
-    governanceDeploymentManager.hre.ethers.provider.on(filter, (log) => {
-      resolve(log);
-    });
-
-    setTimeout(() => {
-      reject(new Error('StateSender.StateSynced event listener timed out'));
-    }, EVENT_LISTENER_TIMEOUT);
+  // grab all events on the StateSender contract since the `startingBlockNumber`
+  const stateSyncedEvents: Event[] = [];
+  const filter = stateSender.filters.StateSynced();
+  const logs = await governanceDeploymentManager.hre.ethers.provider.getLogs({
```

Comment:
I think you just rename `logs` to `stateSyncedEvents`, and then delete lines 60 and 68

```
const stateSyncedEvents: Event[]  = await governanceDeploymentManager.hre.ethers.provider.getLogs({
```

Reply:
Good catch. Done.